packages/jellyfish-testing/src/testing.ts

```diff
@@ -75,12 +87,17 @@ export class TestingGroup {
    */
   static create (
     n: number,
-    init = (index: number) => new MasterNodeRegTestContainer(RegTestFoundationKeys[index])
+    init?: (index: number) => DeFiDContainer
   ): TestingGroup {
     const containers: MasterNodeRegTestContainer[] = []
     const testings: Testing[] = []
+
+    if (init === undefined) {
+      init = (index: number) => new MasterNodeRegTestContainer(RegTestFoundationKeys[index])
+    }
+
     for (let i = 0; i < n; i += 1) {
-      const container = init(i)
+      const container = init(i) as MasterNodeRegTestContainer
```

Comment:
This can get problematic as the `init` method now returns `DeFiDContainer` which may or may not be a `MasterNodeRegTestContainer` which we are casting to at line `#100`. This can lead to incorrect inference when someone is creating a group that does not support the same functionality as a `MasterNode`.

Here is a [reference](https://github.com/DeFiCh/jellyfish/blob/b459b6615b6f040581902253c9055dda9bd9f2b6/packages/jellyfish-testing/src/testing.ts) for when I was running down this rabbit hole.

Reply:
indeed tricky as i was also balancing between not affecting too much exisitng stuff vs some major refactor. but! let me read the reference that you shared

Reply:
just pushed a slight edit to remove `as MasterNodeRegTestContainer`